monasca/templates/api-deployment.yaml

```diff
@@ -51,11 +51,17 @@ spec:
             - name: MYSQL_HOST
               value: "{{ .Release.Name }}-mysql"
             - name: MYSQL_USER
-              value: {{ .Values.api.mysql.user | quote }}
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Release.Name }}-mysql-api-secret"
+                  key: username
             - name: MYSQL_PASSWORD
-              value: {{ .Values.api.mysql.password | quote }}
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Release.Name }}-mysql-api-secret"
+                  key: password
             - name: MYSQL_DB
-              value: {{ .Values.mysql.mysqlDatabase | quote }}
+              value: "mon"
```

Comment:
I just merged hpcloud-mon/monasca-docker#23 to make the MySQL database name configurable.  But, thinking about it... I'm still not sure we want to use the `mysql.mysqlDatabase` property. `mysql-init` creates its own database and users automatically, and will drop any database with the same name.

The mysql chart should handle an unset `mysqlDatabase` sanely, so I'd say we should just not set it. If we want to make the database name configurable, we can do that under a different property name and avoid unnecessarily dropping and recreating the database.